### DISCUSSION TITLE
getSnapshot doesn't return actor's last emitted value

### DISCUSSION THREAD

**Discussion by jikez** [statelyai/xstate#4910] [General]
We use xstate for our user onboarding flow in our react native app, the code is shorten as below: 
```
  "enterPhoneNumber:": {
        entry: () =>
          navigationRef.navigate(ENTER_PHONE_NUMBER_SCREEN),
        on: {
          submit: {
            actions: assign(({ event: { value }, context }) => {
              return { ...context, ...value }
            }),
            target: "phoneVerification"
          },
        },
      },
      "phoneVerification": {
        entry:  navigationRef.navigate(
           PHONE_VERIFICATION_SCREEN
            )
        },
....
      },
```

From ENTER_PHONE_NUMBER_SCREEN, when we hit `actor.send({type: "submit"}, {value: {phoneNumber})`, it triggers the transition from `enterPhoneNumber` to `phoneVerification`, and on entry it navigates users to PHONE_VERIFICATION_SCREEN. This works as expected. 
However, in PHONE_VERIFICATION_SCREEN, when we try to get context from `actor.getSnapShot`, `phoneNumber` is not always populated. For some users it has the phoneNumber that's assigned to the context, for some users it doesn't. 

We also tried to add a guard for the transition like: 
```
 guard: ({ context }) => {
              return context.phoneNumber !== ''
            },
```
but it doesn't work either. 

If we add a slight delay on the navigation like this 
```
   entry: () => {
          setTimeout(() => {
            navigationRef.navigate(
              AppRoute.ONBOARDING_PHONE_VERIFICATION_SCREEN,
            )
          }, 100)
        },
```
it works without any problem. 

My questions are:
- seems like the context in `guard` is the previous context, if that's the case, how can we make sure it only transit based on the latest context?
- `actor.getSnapShot` doesn't seem to have the latest context after transition happens either, what's the correct way to get latest context if we ever want to use that in the component?

**Comment by davidkpiano** [comment]
You have to send it in the same event object:

```ts
actor.send({ type: "submit", value: phoneNumber })
```

**Reply by jikez** [reply]
hi! 
sorry that was a typo, it was actually sent it in the same object as 
```
actor.send({
      type: 'submit',
      value: {
        phoneNumberCode,
        phoneNumber,
      },
    })
```
